McMasterNEUDOSE/cFS
# Creation of Base CSP App

**Describe the contribution**
Created a new app that will facilitate all CFS external communication in place of the to and ci apps.
- This is the base of the app, the functionality to send/receive CFS SB messages via CSP needs to be added
- Currently, the app sends mock data via the CSP loopback interface between server/client CFE child tasks

**Testing performed**
1. Execute `./run.sh`
1. Observe CFE system log
1. Confirmed child tasks were correctly registered
1. Confirmed mock data was sent between the server/client

**Expected behavior changes**
External communication will now be defined in a single application module (csp app) rather than 2 modules (ci/to)

**System(s) tested on**
 - Hardware: PC
 - OS: Ubuntu 18.04

**Additional context**
- I am planning to develop an interface for relaying the packets over UDP to the ground station in place of the CAN Bus interface
- The purpose of this will be to aid in the Linux development process

**Contributor Info - All information REQUIRED for consideration of pull request**
- Stephen Scott, McMaster NEUDOSE
